csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.cs

```diff
@@ -993,9 +998,171 @@ public bool IsTensor
     }
 
 
-    internal class ModelMetadata
+    public class ModelMetadata
     {
-        //TODO: placeholder for Model metadata. Currently C-API does not expose this.
+        private string _producerName;
+        private string _graphName;
+        private string _domain;
+        private string _description;
+        private long _version;
+        private Dictionary<string, string> _customMetadataMap = new Dictionary<string, string>();
+
+        internal ModelMetadata(InferenceSession session)
+        {
+            IntPtr modelMetadataHandle = IntPtr.Zero;
+
+            IntPtr producerNameHandle = IntPtr.Zero;
```

Comment:
>IntPtr producerNameHandle = IntPtr.Zero; [](start = 12, length = 40)

This should go right before OrtModelMetadataGetProducerName() call

Reply:
done